assertion_test.go

```diff
@@ -477,3 +477,183 @@ func TestAssertion_Strings(t *testing.T) {
+			failure: AssertionFailure{
+				Expected: &AssertionValue{},
+			},
+			traits: fieldTraits{
+				Expected: fieldDenied,
+			},
+		},
+		{
+			testName:          "Required value on Expected to be pointer to AssertionRange",
+			errorContainsText: "AssertionValue should contain AssertionRange",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: &AssertionRange{},
+				},
+			},
+			traits: fieldTraits{
+				Range: fieldRequired,
+			},
+		},
+		{
+			testName:          "Required value on Expected to have min on range",
+			errorContainsText: "AssertionRange value should have non-nil Min field",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: AssertionRange{
+						Max: 1,
+					},
+				},
+			},
+			traits: fieldTraits{
+				Range: fieldRequired,
+			},
+		},
+		{
+			testName:          "Required value on Expected to have max on range",
+			errorContainsText: "AssertionRange value should have non-nil Max field",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: AssertionRange{
+						Min: 1,
+					},
+				},
+			},
+			traits: fieldTraits{
+				Range: fieldRequired,
+			},
+		},
+		{
+			testName:          "Required value on Expected to be pointer to AssertionList",
+			errorContainsText: "AssertionValue should contain AssertionList",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: &AssertionList{},
+				},
+			},
+			traits: fieldTraits{
+				List: fieldRequired,
+			},
+		},
+		{
+			testName:          "Required value on Expected to be pointer to AssertionList",
+			errorContainsText: "AssertionValue should contain AssertionList",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: &AssertionList{},
+				},
+			},
+			traits: fieldTraits{
+				List: fieldRequired,
+			},
+		},
+		{
+			testName:          "Required value on Expected AssertionList with len",
+			errorContainsText: "AssertionList should be non-empty",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: AssertionList{},
+				},
+			},
+			traits: fieldTraits{
+				List: fieldRequired,
+			},
+		},
+		{
+			testName:          "Required Expected AssertionList with values not being a slice",
+			errorContainsText: "but it contains a single element which itself is a list",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: AssertionList{[]int{1}},
+				},
+			},
+			traits: fieldTraits{
+				List: fieldRequired,
+			},
+		},
+		{
+			testName:          "Required Expected",
+			errorContainsText: "",
+			failure: AssertionFailure{
+				Expected: nil,
+			},
+			traits: fieldTraits{
+				List: fieldRequired,
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.testName, func(t *testing.T) {
+			err := validateTraits(&test.failure, test.traits)
+			require.Error(t, err)
+			require.Contains(t, err.Error(), test.errorContainsText)
+		})
+	}
+
+	t.Run("nil Errors", func(t *testing.T) {
```

Comment:
Let's rename this to "no error" to avoid confusion with "nil Errors" test in TestAssertion_Validation.